tests/test-cases/Group11-Upgrade/11-03-Upgrade-InsecureRegistry.robot

```diff
@@ -14,25 +14,100 @@
+    ${rc}  ${output}=  Run And Return Rc And Output  echo "From busybox" | ${docker} -H ${default_local_docker_endpoint} build -t ${registry_ip}/test/busybox -
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Successfully built
+    Log To Console  \nbusybox built successfully
+    :FOR  ${i}  IN RANGE  6
+    \   ${rc}  ${output}=  Run And Return Rc And Output  ${docker} -H ${default_local_docker_endpoint} login --username ${registry_user} --password ${registry_password} ${registry_ip}
+    \   ${status}  ${msg}=  Run Keyword And Ignore Error  Should Contain  ${output}  Login Succeeded
+    \   Run Keyword If  '${rc}' == '0' and '${status}' != 'FAIL'  Set Test Variable  ${pass}  2
+    \   Exit For Loop If  '${rc}' == '0' and '${status}' != 'FAIL'
+    \   Sleep  10s
+    Should Be Equal As Integers  ${pass}  2
+    Log To Console  \nLogin successfully
+    ${rc}=  Run And Return Rc  ${docker} -H ${default_local_docker_endpoint} push ${registry_ip}/test/busybox
+    Should Be Equal As Integers  ${rc}  0
+    Log To Console  \nbusybox pushed successfully
+
+Test VCH And Registry
+    [Arguments]  ${vch_endpoint}  ${registry_ip}  ${docker}=DOCKER_API_VERSION=1.23 docker
+    ${rc}  ${output}=  Run And Return Rc And Output  ${docker} -H ${vch_endpoint} pull ${registry_ip}/test/busybox
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Digest:
+    Should Contain  ${output}  Status:
+    Should Not Contain  ${output}  Error response from daemon
+
+Cleanup Test Environment
+    [Arguments]  ${handle}  ${docker_daemon_pid}  ${harbor_name}  ${harbor_ip}  ${docker}=DOCKER_API_VERSION=1.23 docker
+    ${rc}=  Run And Return Rc  ${docker} -H ${default_local_docker_endpoint} rmi ${harbor_ip}/test/busybox
+    Should Be Equal As Integers  ${rc}  0
+    Kill Local Docker Daemon  ${handle}  ${docker_daemon_pid}
+    Cleanup Harbor  ${harbor_name}
+    Clean up VIC Appliance And Local Binary
+
+*** Variables ***
+${test_vic_version}  7315
+${vic_success}  Installer completed successfully
+${docker_bridge_network}  bridge
+${docker_daemon_default_port}  2375
+${http_harbor_name}  integration-test-harbor-http
+${https_harbor_name}  integration-test-harbor-https
+${default_local_docker_endpoint}  unix:///var/run/docker-local.sock
 
 *** Test Cases ***
-Upgrade VCH with InsecureRegistry
-    ${oldHost}  ${oldPath}=  Get host and path from guestinfo
-    Log  ${oldHost} ${oldPath}
-    Should Be Equal As Strings  ${oldHost}  "<nil>"
-    Should Be Equal As Strings  ${oldPath}  "10.10.10.10:1234"
+Upgrade VCH with Harbor On HTTP
+    ${harbor_ip}=  Install Harbor To Test Server  ${http_harbor_name}
+    Add Project On Registry  ${harbor_ip}  http
+    ${handle}  ${docker_daemon_pid}=  Setup Test Environment  ${harbor_ip}
+
+    Install VIC with version to Test Server  ${test_vic_version}  --insecure-registry ${harbor_ip} --no-tls --no-tlsverify
```

Comment:
both no-tls and no-tlsverify are not valid. either one or the other needs to be specified.